### PR TITLE
fix(core): prevent floating sidebar from disappearing unexpectedly

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1475

Fixed an issue where you couldn't create collection or use the rename function in the floating sidebar.

https://github.com/user-attachments/assets/41c2b6a8-8fc9-4f8b-ab51-bd7ce2a58738

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#8477** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8477?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pengx17 — COMMENTED** [review]
Thoughts: Normally for react portals that are in the sub tree of appsidebar, there `mouseleave` handler should not trigger moving from the sidebar to nested portals, e.g. the operation menus. However some of our implementations of *modals* are either using `useMount` to mount the modal outside of the normal react tree, or being rendered somewhere globally.

Determining whether the mouseleave event should be handled without a defined React tree is tricky. In my opinion, you should add condition checks in `onMouseLeave`, rather than in the useEffect hook.

A working example
```ts
  const onMouseLeave = useCallback(
    (e: React.MouseEvent) => {
      const target = e.relatedTarget as HTMLElement;
      if (
        target?.closest(
          '[role="dialog"], [role="menu"], modal-transition-container'
        )
      ) {
        return;
      }
      appSidebarService.setHovering(false);
    },
    [appSidebarService]
  );
```

**Comment by JimmFly** [comment]
> Thoughts: Normally for react portals that are in the sub tree of appsidebar, there `mouseleave` handler should not trigger moving from the sidebar to nested portals, e.g. the operation menus. However some of our implementations of _modals_ are either using `useMount` to mount the modal outside of the normal react tree, or being rendered somewhere globally.
> 
> Determining whether the mouseleave event should be handled without a defined React tree is tricky. In my opinion, you should add condition checks in `onMouseLeave`, rather than in the useEffect hook.


It seems that it doesn't work when I want to use the rename function. @pengx17 

CleanShot 2024-10-11 at 14 56 39@2x

**Comment by pengx17** [comment]
add the following check should work `!(target instanceof HTMLElement) || resizing ||`
i think this is because radix tries to blur the element and as a result the window object got focused

**Comment by pengx17** [comment]
I also noticed that resizing sidebar becomes slow. Can you take a look?

**Comment by JimmFly** [comment]
> I also noticed that resizing sidebar becomes slow. Can you take a look?

Does it happen on pages other than edgeless page?

**Comment by pengx17** [comment]
happens everywhere

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/modules/app-sidebar/views/index.tsx [hunk 1]
```diff
@@ -128,9 +128,33 @@ export function AppSidebar({ children }: PropsWithChildren) {
...
+      ) {
+        return;
+      }
+
+      // handle uncaught popper
+      const popper = document.querySelector(
+        '[data-radix-popper-content-wrapper]'
+      );
+      if (popper) {
+        return;
```
Comment (EYHN) [inline packages/frontend/core/src/modules/app-sidebar/views/index.tsx]: It seems that in this case the sidebar never closes, even if the dialog has been closed.

Comment (EYHN) [reply packages/frontend/core/src/modules/app-sidebar/views/index.tsx]: I think we can try, when the sidebar is floating, we listen to the mousemove event on the window, and close it when the mouse moves a certain distance out of the sidebar range

**Pull request merged**